### ISSUE TITLE
1.0.0 and stabilizing the API?

### ISSUE THREAD

**Issue by JamesMcMahon** [sschmid/Entitas-CSharp#431]
I figured this was a good follow up to the [big recent announcement](https://github.com/sschmid/Entitas-CSharp/issues/427). Now that Simon is going to have more time to work on Entitas are there any plans for a 1.0.0 release?

I think the benefits for the Entitas community would be:

- Stabilize the API to avoid the overhead of breaking changes 
- Use [Semantic Versioning](http://semver.org/) to communicate out breaking changes (as needed)
- Have the community be able to write example projects, documentation and tutorials based on a stable API would allow us to grow the community

Personally, as someone who just spent a decent chunk of time updating [my example project](https://github.com/JamesMcMahon/entitas-2d-roguelike) up to the latest release, I think the API is a really good place and the focus now could be on non-breaking features and community building.

Thoughts?

**Comment by sschmid** [comment]
Hi!

v 1.0 definitely hit my mind several times when thinking about the Asset Store release. Personally, I want to have a better wiki, more tutorials and maybe a website coming along with a 1.0 release to minimize confusion and help people getting started better.

I agree, the API looks pretty stable to me. But going 1.0 won't actually change much about the way I work.

But maybe I just suffer from
https://youtu.be/tc2UgG5L7WM?t=693

**Comment by c0ffeeartc** [comment]
Entitas makes bold changes, maybe its versioning model can change too https://medium.com/sapioit/why-having-3-numbers-in-the-version-name-is-bad-92fc1f6bc73c

**Comment by JamesMcMahon** [comment]
I would't get too hung over the specific versioning to use. At the end of the day versioning is a tool for communicating with your end users.

The big thing is to get to stable version and reduce the number of API breaking releases. I think it will be difficult to build the Entitas community without doing this.

For example, here is what the state of [example projects on the wiki](https://github.com/sschmid/Entitas-CSharp/wiki/Example-projects) currently looks like:

| Name  | Entitas Version |
| ------------- | ------------- |
| Basic Example  | 0.37.0  |
| Match One  | 0.41.2  |
| Shmup  | 0.35.0 |
| Tanks!  | 0.31.1 |
| Unity 2d Roguelike  | 0.42.3 |
| Pong  | 0.31.1 |
| Asteroids  | 0.24.2 |

Because the APIs have changed so much it's difficult to build a corpus of example projects that reflect the current state of Entitas. This also applies to blog post, articles, talks, etc. For instance I got this comment on the talk I gave on Entitas / ECS:

<img width="625" alt="screen shot 2017-06-20 at 5 34 46 pm" src="https://user-images.githubusercontent.com/916676/27358949-a58981fa-55de-11e7-889d-7c44176ffe5f.png">

Basically I think any project needs to balance implementing the next great idea with user churn and I think the churn is high right now for Entitas, which makes sense as Simon as coming up with cool ideas left and right. But at some point the shit should be made to reduce churn. 

I think Entitas has a great like community and I am eager to see it grow as the API stabilizes.

**Comment by jonathanpmast** [comment]
As someone that is just coming to this project, and generally has a pretty sound understanding of the ECS architecture.  This issue is extremely relevant to me :)

Trying to follow [the youtube tutorials](https://youtu.be/0wodRUOoHKg?t=647) and the API has changed enough that I've had to cross reference the sample projects linked in this thread to make progress.

Not impossible, and given a community supported lib I guess we kind of expect it, but I think @JamesMcMahon feedback is super relevant here.
